cmd/kubeconform/main.go

```diff
@@ -1,337 +1,94 @@
+			if err != nil {
 				fmt.Fprint(os.Stderr, "failed writing log\n")
 			}
 		}
-	}
-
-	result <- success
-}
-
-func getFiles(files []string, fileBatches chan []string, validationResults chan []validationResult) {
-	for _, filename := range files {
-		file, err := os.Open(filename)
-		if err != nil {
-			validationResults <- []validationResult{{
-				filename: filename,
-				err:      err,
-				skipped:  false,
-			}}
-			continue
-		}
-		defer file.Close()
-
-		fi, err := file.Stat()
-		switch {
-		case err != nil:
-			validationResults <- []validationResult{{
-				filename: filename,
-				err:      err,
-				skipped:  false,
-			}}
-
-		case fi.IsDir():
-			if err := fsutils.FindYamlInDir(filename, fileBatches, 10); err != nil {
-				validationResults <- []validationResult{{
-					filename: filename,
-					err:      err,
-					skipped:  false,
-				}}
-			}
+		done <- ret
+	}()
 
-		default:
-			fileBatches <- []string{filename}
-		}
-	}
+	return done
 }
 
-func realMain() int {
-	var schemaLocationsParam arrayParam
-	var skipKindsCSV, k8sVersion, outputFormat string
-	var summary, strict, verbose, ignoreMissingSchemas, help bool
-	var nWorkers int
-	var err error
-	var files []string
-
-	flag.StringVar(&k8sVersion, "kubernetes-version", "1.18.0", "version of Kubernetes to validate against")
-	flag.Var(&schemaLocationsParam, "schema-location", "override schemas location search path (can be specified multiple times)")
-	flag.BoolVar(&ignoreMissingSchemas, "ignore-missing-schemas", false, "skip files with missing schemas instead of failing")
-	flag.BoolVar(&summary, "summary", false, "print a summary at the end")
-	flag.IntVar(&nWorkers, "n", 4, "number of routines to run in parallel")
-	flag.StringVar(&skipKindsCSV, "skip", "", "comma-separated list of kinds to ignore")
-	flag.BoolVar(&strict, "strict", false, "disallow additional properties not in schema")
-	flag.StringVar(&outputFormat, "output", "text", "output format - text, json")
-	flag.BoolVar(&verbose, "verbose", false, "print results for all resources")
-	flag.BoolVar(&help, "h", false, "show help information")
-	flag.Usage = func() {
-		fmt.Fprintf(os.Stderr, "Usage: %s [OPTION]... [FILE OR FOLDER]...\n", os.Args[0])
-		flag.PrintDefaults()
-	}
-
-	flag.Parse()
-
-	if help {
-		flag.Usage()
-		return 1
-	}
-
-	skipKinds := skipKindsMap(skipKindsCSV)
-
-	for _, file := range flag.Args() {
-		files = append(files, file)
-	}
-
-	filter := func(signature resource.Signature) bool {
-		isSkipKind, ok := skipKinds[signature.Kind]
-		return ok && isSkipKind
+func main() {
+	conf := config.FromFlags()
+	if conf.Help {
+		os.Exit(0)
 	}
 
-	if len(schemaLocationsParam) == 0 {
-		schemaLocationsParam = append(schemaLocationsParam, "https://kubernetesjsonschema.dev") // if not specified, default behaviour is to use kubernetesjson-schema.dev as registry
+	if conf.NumberOfWorkers == -1 {
+		conf.NumberOfWorkers = runtime.NumCPU()
```

Comment:
Not sure if this is the best, in this case the limiting factor is often downloading the schemas, which is not very CPU intensive - so one 1-2 cores (usual in CI) you d still benefit from more workers